test/gtest/common/mem_buffer.cc

```diff
@@ -83,7 +85,14 @@ const std::vector<ucs_memory_type_t>&  mem_buffer::supported_mem_types()
         }
         if (is_rocm_supported()) {
             vec.push_back(UCS_MEMORY_TYPE_ROCM);
-            vec.push_back(UCS_MEMORY_TYPE_ROCM_MANAGED);
+
+            if (hipGetDevice(&deviceId) == hipSuccess &&
+                hipDeviceGetAttribute(&hasManagedMem,
+                                      hipDeviceAttributeManagedMemory,
+                                      deviceId) == hipSuccess &&
+                hasManagedMem != 0) {
+                vec.push_back(UCS_MEMORY_TYPE_ROCM_MANAGED);
+            }
```

Comment:
maybe add public function 
```
bool mem_buffer::is_rocm_managed_supported() 
{
    int device_id, has_managed_mem;
    return ((hipGetDevice(&device_id) == hipSuccess) && 
            (hipDeviceGetAttribute(&has_managed_mem,
                                   hipDeviceAttributeManagedMemory,
                                   deviceId) == hipSuccess) &&
            has_managed_mem;
}
```

then use it in `mem_buffer::supported_mem_types()` and `UCS_TEST_F(rocm_hooks, test_hipMallocManaged) {`

Reply:
That would be a lot cleaner.  Thanks for these suggestions.